### PR TITLE
Nodejs/week2

### PR DESCRIPTION
**Pull request by hnnhack**

### CODE REVIEW

File: week2/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,56 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const program = require('commander');
```
Comment (gajduk) [inline week2/homework/src/index.js]: commander 👍

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+    .description('Removes all todo items')
+    .action(() => {
+      data.resetTodo();
+      data.listTodo();
+    });
+
+  program
+    .command('update')
+    .description('Updates a todo by index')
+    .action((index, todo) => {
```
Comment (gajduk) [inline week2/homework/src/index.js]: nice and clean

File: week2/homework/src/todo.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  }
+};
+
+const fetchData = () => {
+  try {
+    const dataString = fs.readFileSync('todoData.json');
+    return JSON.parse(dataString);
+  }
+ catch (error) {
+    return [];
```
Comment (gajduk) [inline week2/homework/src/todo.js]: 😞 
If you are going to ignore an error add a comment to explain to other developers why

File: week2/homework/src/todo.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    const isDuplicate = todoArray.filter(todo => todo.title === title);
+    if (isDuplicate.length === 0) {
+      todoArray.push(todo);
+      saveData(todoArray);
+      console.log('Todo item added');
+      console.log('---------------');
+      return todo;
+    }
+ else {
+      console.log('Todo already exists');
```
Comment (gajduk) [inline week2/homework/src/todo.js]: error handling 👍

File: week2/homework/src/todo.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  }
+ catch (error) {
+    return console.log(error.message);
+  }
+};
+
+const deleteTodo = index => {
+  try {
+    if (index > 0) {
+      const isFiltered = todoArray.filter(todo => todo.title !== todoArray[index].title);
```
Comment (gajduk) [inline week2/homework/src/todo.js]: interesting approach. maybe it is simpler to write `todoArray.splice(index,1);`

File: week2/homework/src/todo.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    return console.log(error.message);
+  }
+};
+
+const saveData = data => {
+  try {
+    fs.writeFileSync('todoData.json', JSON.stringify(data));
+  }
+ catch (error) {
+    return console.log(error.message);
```
Comment (gajduk) [inline week2/homework/src/todo.js]: not that it is wrong but is it really necessary to catch the error here? Would it not be caught in updateTodo for example

File: week2/homework/src/todo.js (hunk 3 above)
Comment (hnnhack) [reply week2/homework/src/todo.js]: I have read an article about rethrow errors, I have just use it in my homework to know does it useful or not.

File: week2/homework/src/index.js (hunk 2 above)
Comment (hnnhack) [reply week2/homework/src/index.js]: Thank you very much!

File: week2/homework/src/index.js (hunk 1 above)
Comment (hnnhack) [reply week2/homework/src/index.js]: Thanks

File: week2/homework/src/todo.js (hunk 4 above)
Comment (hnnhack) [reply week2/homework/src/todo.js]: thanks

File: week2/homework/src/todo.js (hunk 5 above)
Comment (hnnhack) [reply week2/homework/src/todo.js]: Yeah, it is simple way,.. thanks.

File: week2/homework/src/todo.js (hunk 6 above)
Comment (hnnhack) [reply week2/homework/src/todo.js]: Yes you are right. No need to use here,..

**Comment by hnnhack** [comment]
Thank you very much @gajduk, your extra efforts for reviewing my homework, for your courses and valuable tricks. Wish you the best!

**Pull request closed without merging**